diff -urN linux-xlnx-xilinx-v2015.4.01.orig/drivers/misc/eeprom/at24.c linux-xlnx-xilinx-v2015.4.01/drivers/misc/eeprom/at24.c
--- linux-xlnx-xilinx-v2015.4.01.orig/drivers/misc/eeprom/at24.c	2016-04-30 10:31:56.526948078 +0900
+++ linux-xlnx-xilinx-v2015.4.01/drivers/misc/eeprom/at24.c	2016-04-30 10:37:06.154529775 +0900
@@ -519,6 +519,9 @@
 		 * is recommended anyhow.
 		 */
 		chip.page_size = 1;
+		if(!strncmp(id->name,"24c02",5)){
+			chip.page_size = 8;
+		}
 
 		/* update chipdata if OF is present */
 		at24_get_ofdata(client, &chip);
